pkg/api/registered/registered.go

```diff
@@ -25,71 +26,97 @@ import (
 	"k8s.io/kubernetes/pkg/api/unversioned"
 )
 
-// List of registered API versions.
-// The list is in the order of most preferred to the least.
-var RegisteredGroupVersions []unversioned.GroupVersion
+var (
+	// registeredVersions represents all enabled API versions. Please call
+	// RegisterVersions() to register the versions.
+	registeredVersions = map[unversioned.GroupVersion]struct{}{}
```

Comment:
@deads2k sorry about the godoc, I should have updated it after a grep&sed.

I'm thinking probably we don't need the `registeredVersion` we talked about in the call. We used to need it because we compare KUBE_API_VERSIONS with `registeredVersion` to make sure the versions in the env variable makes sense, but we have another mechanism to check that now.

What do you think? Why do we need to keep record of the versions that are compiled but not enabled?